### PR TITLE
Apply delay on send method

### PR DESCRIPTION
**Pull request by HUG-socloz**

### CODE REVIEW

File: src/Messenger/Transport/AzureTransport.php [hunk 1]
```diff
@@ -161,6 +162,14 @@ public function send(Envelope $envelope): Envelope
         // Build broker properties
         /** @var null|AzureBrokerPropertiesStamp $brokerProperties */
         $brokerProperties = $envelope->last(AzureBrokerPropertiesStamp::class);
+
+        /** @var null|DelayStamp $delayStamp */
+        $delayStamp =  $envelope->last(DelayStamp::class);
+        if ($delayStamp) {
+            $now = new \DateTime('UTC');
+            $brokerProperties->setScheduledEnqueueTimeUtc((clone $now)->modify(sprintf('+%d seconds', $delayStamp->getDelay()/1000)));
```
Comment (AymDev) [inline src/Messenger/Transport/AzureTransport.php:170]: The `$brokerProperties` may be null, it should be created if needed.

File: tests/Messenger/Transport/AzureTransportTest.php [hunk 2]
```diff
@@ -553,6 +554,44 @@ public function testSendWithBrokerPropertiesStampGetsConvertedToHttpHeader(): vo
         $transport->send($envelope);
     }
 
+    /**
+     * The BrokerProperties stamp must be used to generate a BrokerProperties HTTP header
+     */
+    public function testSendWithBrokerPropertiesStampAndDelayStamp(): void
+    {
+        $envelope = new Envelope(new class {}, [
+            new AzureBrokerPropertiesStamp(), new DelayStamp(50000)
```
Comment (AymDev) [inline tests/Messenger/Transport/AzureTransportTest.php:563]: The `AzureBrokerPropertiesStamp` should be removed here as it should automatically be added.

File: tests/Messenger/Transport/AzureTransportTest.php [hunk 3]
```diff
@@ -553,6 +554,44 @@ public function testSendWithBrokerPropertiesStampGetsConvertedToHttpHeader(): vo
...
+        $serializer = self::createMock(SerializerInterface::class);
+        $serializer->expects(self::once())
+            ->method('encode')
+            ->with($envelope)
+            ->willReturn(['body' => 'test-body']);
+
+        $sender = new MockHttpClient(function (string $method, string $url, array $options): ResponseInterface {
+            self::assertArrayHasKey('normalized_headers', $options);
+            self::assertArrayHasKey('brokerproperties', $options['normalized_headers']);
+            self::assertSame('BrokerProperties: {}', $options['normalized_headers']['brokerproperties'][0]);
```
Comment (AymDev) [inline tests/Messenger/Transport/AzureTransportTest.php:575]: Wouldn't this test fail here ? I'd guess the header would be something like `BrokerProperties: {"ScheduledEnqueueTimeUtc": 123456789}`

**Comment by AymDev** [comment]
Thanks for this PR, I'll release the v1.4 in a few minutes.

**Comment by HUG-socloz** [comment]
Pleased to participate.

**Pull request merged**